clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleSecurityGroupProvider.groovy

```diff
@@ -216,6 +216,15 @@ class GoogleSecurityGroupProvider implements SecurityGroupProvider<GoogleSecurit
         targetServiceAccountsInCommon = (securityGroup.targetServiceAccounts).intersect(serviceAccounts)
       }
 
+      /** Due to how shared VPC projects share firewall rules, they look "duplicated" in Spinnaker and show
+      * once per account. This derives the project the firewall is hosted in, and confirms that it matches where the
+      * Security group is located.
+      */
+      def firewallProjectAndAccountMatch = GCEUtil.deriveProjectId(securityGroup.selfLink) == account
```

Comment:
I don't think `projectId` and `account` are equivalent here.

My understanding here is that the point of VPC is to manage network resources in one project called the 'host project' and then utilize those network resources in a separate project called the 'service project', so the two are segregated as far as permissions are concerned. I think the duplication comes from the [caching agent](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSecurityGroupCachingAgent.groovy) writing the duplicate shared VPC security groups to redis for each configured GCE account, so this is likely the problem.